cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectConfigurationBuilder.java

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model;
+
+
+import io.strimzi.api.kafka.model.common.ClientTls;
+import io.strimzi.api.kafka.model.common.authentication.KafkaClientAuthentication;
+import io.strimzi.api.kafka.model.common.authentication.KafkaClientAuthenticationOAuth;
+import io.strimzi.api.kafka.model.common.authentication.KafkaClientAuthenticationPlain;
+import io.strimzi.api.kafka.model.common.authentication.KafkaClientAuthenticationScram;
+import io.strimzi.api.kafka.model.common.authentication.KafkaClientAuthenticationScramSha256;
+import io.strimzi.api.kafka.model.common.authentication.KafkaClientAuthenticationScramSha512;
+import io.strimzi.api.kafka.model.common.authentication.KafkaClientAuthenticationTls;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+import static io.strimzi.operator.cluster.model.KafkaConnectCluster.PASSWORD_VOLUME_MOUNT;
+
+/**
+ * This class is used to generate the Connect configuration template. The template is later passed using a config map to
+ * the connect pods. The scripts in the container images will fill in the variables in the template and use the
+ * configuration file. This class is using the builder pattern to make it easy to test the different parts etc. To
+ * generate the configuration file, it is using the PrintWriter.
+ */
+public class KafkaConnectConfigurationBuilder {
+    // Names of environment variables expanded through config providers inside Connect node
+    private final static String PLACEHOLDER_CERT_STORE_PASSWORD_CONFIG_PROVIDER_ENV_VAR = "${strimzienv:CERTS_STORE_PASSWORD}";
+    private final static String PLACEHOLDER_SASL_USERNAME_CONFIG_PROVIDER_ENV_VAR = "${strimzienv:KAFKA_CONNECT_SASL_USERNAME}";
+    // the SASL password file template includes: <volume_mount>/<secret_name>/<password_file>
+    private static final String PLACEHOLDER_SASL_PASSWORD_FILE_TEMPLATE_CONFIG_PROVIDER_DIR = "${strimzidir:%s%s:%s}";
+
+    private final StringWriter stringWriter = new StringWriter();
+    private final PrintWriter writer = new PrintWriter(stringWriter);
+    private String securityProtocol = "PLAINTEXT";
+
+    /**
+     * Connect configuration template constructor
+     *
+     * @param bootstrapServers  Kafka cluster bootstrap servers to connect to
+     */
+    public KafkaConnectConfigurationBuilder(String bootstrapServers) {
+        printHeader();
+        configureBootstrapServers(bootstrapServers);
+    }
+
+    /**
+     * Renders the Kafka cluster bootstrap servers configuration
+     *
+     * @param bootstrapServers  Kafka cluster bootstrap servers to connect to
+     */
+    private void configureBootstrapServers(String bootstrapServers) {
```

Comment:
```suggestion
    private void printBootstrapServers(String bootstrapServers) {
```
There's a mixture of method names, should we fix on `with..` for the public methods and `print...` for the internal ones?